sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/CombineFns.java

```diff
@@ -994,28 +994,28 @@ public static ComposedAccumulatorCoder of(
     public void encode(Object[] value, OutputStream outStream, Context context)
         throws CoderException, IOException {
       checkArgument(value.length == codersCount);
+      if (value.length == 0) {
+        return;
+      }
       Context nestedContext = context.nested();
-      for (int i = 0; i < codersCount; ++i) {
-        if (i < codersCount - 1) {
-          coders.get(i).encode(value[i], outStream, nestedContext);
-        } else {
-          coders.get(i).encode(value[i], outStream, context);
-        }
+      for (int i = 0; i < codersCount - 1; ++i) {
```

Comment:
I like how you used the place holder variable to store the `lastIndex` in CoGbkResult.java, would recommend using it here and in `decode`